tools/generator/templates/player/cs/interface.cc.jinja2

```diff
@@ -38,6 +39,32 @@ class CSharpInterface
         MonoObject* _object;
 };
 
+// Static mapping between C++ types and mono_get_*_class.
+// Usage: MonoClassFactory<double>::get_class -> mono_get_double_class
+
+template <typename T>
+struct MonoClassFactory
+{
+    static const std::function<MonoClass*()> get_class;
+};
+
+template <typename T>
+const std::function<MonoClass*()> MonoClassFactory<T>::get_class =
+    T::__if_that_triggers_an_error_there_is_a_problem;
+
+#define MAP_TYPE_MONOCLASS(CxxType, Factory) \
+template <> \
+const std::function<MonoClass*()> MonoClassFactory<CxxType>::get_class = \
+    Factory;
+
+MAP_TYPE_MONOCLASS(int, mono_get_int32_class);
+MAP_TYPE_MONOCLASS(double, mono_get_double_class);
+MAP_TYPE_MONOCLASS(bool, mono_get_boolean_class);
+MAP_TYPE_MONOCLASS(std::string, mono_get_string_class);
+
+#undef MAP_TYPE_MONOCLASS
```

Comment:
Could something like
```c++
// Usage: get_mono_class<double>() -> mono_get_double_class()
template <typename T>
MonoClass* get_mono_class()
{
    if constexpr (std::is_same_v<T, int>)
        return mono_get_int32_class();
    else if constexpr (std::is_same_v<T, double>)
        return mono_get_double_class();
    else if constexpr (std::is_same_v<T, bool>)
        return mono_get_boolean_class();
    else if constexpr (std::is_same_v<T, std::string>)
        return mono_get_string_class();
    else
        return T::__if_that_triggers_an_error_there_is_a_problem;
}
```
be enough? It's shorter and less convoluted.

I successfully compiled a C# champion with this version but didn't try executing it. I also tried using this function with an unsuported type and it fails properly at compile-time when trying to compile the content of the else.